### PR TITLE
feat: adds upgrade test for azure

### PR DESCRIPTION
**Pull request by nilekhc**
Signed-off-by: Nilekh Chaudhari <1626598+nilekhc@users.noreply.github.com>

This PR adds periodic job for e2e azure upgrade test.

### CODE REVIEW

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml [hunk 1]
```diff
@@ -592,3 +592,35 @@ periodics:
...
+        - bash
+        - -c
+        - >-
+          HELM_CHART_DIR=$(ls -h charts/*.tgz | sort --version-sort --field-separator=- --key=5 | tail -n 1) make e2e-bootstrap e2e-helm-deploy-release e2e-azure && HELM_CHART_DIR=manifest_staging/charts/secrets-store-csi-driver make e2e-helm-upgrade e2e-azure
+        # docker-in-docker needs privileged mode
+        securityContext:
+          privileged: true
+  annotations:
+    testgrid-dashboards: sig-auth-secrets-store-csi-driver
+    testgrid-tab-name: secrets-store-csi-driver-image-scan
```
Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:623]:
```suggestion
    testgrid-tab-name: secrets-store-csi-driver-upgrade-test-azure
```

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml [hunk 2]
```diff
@@ -592,3 +592,35 @@ periodics:
...
+    testgrid-alert-email: kubernetes-secrets-store-csi-driver@googlegroups.com
+    description: "Run e2e-azure upgrade test for Secrets Store CSI driver."
```
Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:625]:
```suggestion
    description: "Run driver upgrade test with azure provider for Secrets Store CSI driver."
```

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml [hunk 3]
```diff
@@ -592,3 +592,35 @@ periodics:
     testgrid-alert-email: kubernetes-secrets-store-csi-driver@googlegroups.com
     description: "Run vulnerability scans for Secrets Store CSI driver images."
     testgrid-num-columns-recent: '30'
+
+- interval: 24h
+  name: periodic-secrets-store-csi-driver-upgrade-test-azure
+  decorate: true
+  decoration_config:
+    timeout: 30m
+  labels:
```
Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:601]: These are the required labels

https://github.com/kubernetes/test-infra/blob/f668b08714d2974ff1fbe68302050d74e4bbedf4/config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml#L140-L146

**Review by aramase — COMMENTED** [review]
/lgtm

**Comment by nilekhc** [comment]
Increased timeout for `pull-secrets-store-csi-driver-e2e-windows` based on https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/619

Timeout seems to be consistent in other [PRs](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_secrets-store-csi-driver/602/pull-secrets-store-csi-driver-e2e-windows/1412827141879369728) on the same job.

**Pull request merged**